completion.go

```diff
@@ -0,0 +1,61 @@
+package cli
+
+import (
+	_ "embed"
+	"fmt"
+)
+
+const (
+	completionCommandName = "generate-completion"
+)
+
+//go:embed autocomplete/bash_autocomplete
+var bash_autocomplete string
+
+//go:embed autocomplete/powershell_autocomplete.ps1
+var powershell_autocomplete string
+
+//go:embed autocomplete/zsh_autocomplete
+var zsh_autocomplete string
+
+type renderCompletion func(a *App) (string, error)
+
+func getCompletion(s string) renderCompletion {
+	return func(a *App) (string, error) {
+		return s, nil
+	}
+}
+
+var shellCompletions = map[string]renderCompletion{
+	"bash": getCompletion(bash_autocomplete),
+	"ps":   getCompletion(powershell_autocomplete),
+	"zsh":  getCompletion(zsh_autocomplete),
+	"fish": func(a *App) (string, error) {
+		return a.ToFishCompletion()
+	},
+}
```

Comment:
If you want to keep the file names closely related to what's in the source code, instead of having multiple `snake_case` variable names, you could use an `embed.FS` and reference the files by full name (?)

```suggestion
var (
	//go:embed autocomplete
	autoCompleteFS embed.FS

	shellCompletions = map[string]renderCompletion{
		"bash": getCompletion("autocomplete/bash_autocomplete"),
		"ps":   getCompletion("autocomplete/powershell_autocomplete.ps1"),
		"zsh":  getCompletion("autocomplete/zsh_autocomplete"),
		"fish": func(a *App) (string, error) {
			return a.ToFishCompletion()
		},
	}
)

type renderCompletion func(a *App) (string, error)

func getCompletion(s string) renderCompletion {
	return func(a *App) (string, error) {
		b, err := autoCompleteFS.ReadFile(s)
		return string(b), err
	}
}
```